src/main/java/org/openpnp/model/Placement.java

```diff
@@ -99,7 +109,40 @@ public void setPart(Part part) {
         this.part = part;
         firePropertyChange("part", oldValue, part);
     }
+    
+    public Status getPlacementStatus() {
+    	if (getPart() == null) {
+        	setPlacementStatus(Placement.Status.MissingPart);
+            return Placement.Status.MissingPart;
+        }
+        if (getType() == Placement.Type.Place) {
+            boolean found = false;
+            for (Feeder feeder : Configuration.get().getMachine().getFeeders()) {
+                if (feeder.getPart() == getPart() && feeder.isEnabled()) {
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+            	setPlacementStatus(Placement.Status.MissingFeeder);
+                return Placement.Status.MissingFeeder;
+            }
+
+            if (getPart().getHeight().getValue() == 0) {
+            	setPlacementStatus(Placement.Status.ZeroPartHeight);
+                return Placement.Status.ZeroPartHeight;
+            }
+        }
+        setPlacementStatus(Placement.Status.Ready);
+        return Placement.Status.Ready;
+    }
 
+    public void setPlacementStatus(Status PlacementStatus) {
+    	Status oldValue = this.placementStatus;
+        this.placementStatus = PlacementStatus;
+        firePropertyChange("PlacementStatus", oldValue, PlacementStatus);
```

Comment:
Property name should be placementStatus, not PlacementStatus, to adhere to Java naming standards. Likewise the incoming parameter should be named the same.

Reply:
done